tests/flow/test_foreach.py

```diff
@@ -501,3 +510,78 @@ def test10_edge_manipulation(self):
         res = graph.query("MATCH (n) FOREACH(i in [1] | DELETE n)")
         self.env.assertEquals(res.nodes_deleted, 2)
         self.env.assertEquals(res.relationships_deleted, 1)
+
+    def test11_path_reference(self):
+        """reference a named path in the FOREACH clause"""
+
+        # clean db
+        self.env.flush()
+        graph = Graph(self.env.getConnection(), GRAPH_ID)
+
+        # create some data
+        res = graph.query(
+            """
+            CREATE (:Actor{id:'Marketing'}),
+                   (:Actor{id:'vietld'}),
+                   (:Actor{id:'phuongdd'})
+            """
+        )
+
+        # make sure three relationships was created
+        self.env.assertEquals(res.nodes_created, 3)
+
+        res = graph.query(
+            """
+            MATCH path = (u:Actor), (g:Actor { id: 'Marketing' })
+            WHERE u.id IN ['vietld', 'phuongdd']
+            FOREACH (n IN nodes(path) |
+                CREATE (n)-[:belong]->(g)
+            )
+            """
+        )
+
+        # make sure two relationships was created
+        self.env.assertEquals(res.relationships_created, 2)
+
+        res = graph.query(
+            """
+            MATCH path = (u:Actor), (g:Actor { id: 'Marketing' })
+            WHERE u.id = 'vietld' OR u.id = 'phuongdd'
+            FOREACH (n IN CASE WHEN u.id = 'vietld' THEN nodes(path) ELSE [] END |
+                CREATE (n)-[:belong]->(g)
+            )
+            """
+        )
+
+        # make sure one relationship was created
+        self.env.assertEquals(res.relationships_created, 1)
+
+        # reference the named path inside the FOREACH body
+        res = graph.query(
+            """
+            MATCH path = (u: Actor)
+            FOREACH (n IN CASE WHEN u.id = 'vietld' THEN nodes(path) ELSE [] END |
+                MERGE (:Actor {id: nodes(path)[0].id})
+            )
+            """
+        )
+
+        # make sure no nodes or properties were created
+        self.env.assertEquals(res.nodes_created, 0)
+        self.env.assertEquals(res.properties_set, 0)
+
+        # define the named path in the FOREACH clauses, and refer them
+        res = graph.query(
+            """
+            FOREACH (n in [1] |
+                MERGE p = (a:Actor {id: 'vietld'})
+                FOREACH (node in nodes(p) |
+                    MERGE (:Actor {id: 'vietld'})
+                )
+            )
+            """
```

Comment:
Please modify this query in such a way which will guarantee both loops had been executed, should be reflected in result-set statistics